python/tk_multi_loader/dialog.py

```diff
@@ -1265,6 +1269,110 @@ def _load_entity_presets(self):
         # data has properly arrived in the model.
         self._on_home_clicked()
 
+    def _setup_hierarchy_model(self, app, root):
+        """
+        Create the model and proxy model required by a hierarchy type configuration setting.
+
+        :param app: :class:`Application`, :class:`Engine` or :class:`Framework` bundle instance
+                    associated with the loader.
+        :param root: The path to the root of the Shotgun hierarchy to display.
+        :return: Created `(model, proxy model)`.
+        """
+
+        # Resolve any magic tokens in the root.
+        if app.context.project:
+            project_id = app.context.project.get("id")
+            root = root.replace("{context.project.id}", str(project_id))
+        else:
+            root = None
+
+        # Construct the hierarchy model.
+        model = shotgun_model.SimpleShotgunHierarchyModel(self, bg_task_manager=self._task_manager)
+
+        # Create a proxy model.
+        proxy_model = QtGui.QSortFilterProxyModel(self)
+        proxy_model.setSourceModel(model)
+
+        # Impose and keep the sorting order on the default display role text.
+        proxy_model.sort(0)
+        proxy_model.setDynamicSortFilter(True)
+
+        # The field manager handles retrieving widgets for Shotgun field types.
+        # It needs time to initialize itself, after which the widgets can begin to be populated.
+        # "PublishedFile.entity" means build a hierarchy that leads to
+        # entities that are linked via the PublishedFile.entity field.
+        entity_fields = {"Asset": [ # Fields usually added by the configuration setting.
```

Comment:
After investigation to understand what is possible, I updated the code and commented:

`# We need to provide a dictionary that identifies what additional fields to include`
`# for the loaded hierarchy leaf entities in addition to "id" and "type".`
`# When available, these fields will be used to add info in the detail panel.`

Ideally, the entity type list for entities with publishes should be retrieved from the settings.